File: 프로그래머스/4/49995. 쿠키 구입/쿠키 구입.java
import java.util.*;

class Solution {
    public int solution(int[] cookie) {
        
        int answer = 0;
        
        for(int i=0; i<cookie.length-1; i++){
            int left = i;
            int leftSum = cookie[left];
            int right = i+1;
            int rightSum = cookie[right];
            while(true){
                if(leftSum == rightSum && leftSum > answer){
                    answer = leftSum;
                }else if(leftSum > rightSum && right < cookie.length -1){
                    right++;
                    rightSum += cookie[right];
                }else if(rightSum >= leftSum && left > 0){
                    left--;
                    leftSum += cookie[left];
                }else break;
            }
        }

        return answer;
    }
}

File: 프로그래머스/4/49995. 쿠키 구입/쿠키 구입.js
function solution(cookie) {
    
    var answer = 0;
    
    for(let i=0; i<cookie.length-1; i++){
        var left = i;
        var leftSum = cookie[left];
        var right = i+1;
        var rightSum = cookie[right];
        while(true){
            if(leftSum == rightSum && leftSum > answer){
                answer = leftSum;
            }else if(leftSum > rightSum && right < cookie.length-1){
                right++;
                rightSum += cookie[right];
            }else if(leftSum <= rightSum && left > 0){
                left--;
                leftSum += cookie[left];
            }else break;
        }
    }
    
    return answer;
}